app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/SessionsAdapter.kt

```diff
@@ -0,0 +1,232 @@
+package nerd.tuxmobil.fahrplan.congress.base
+
+import nerd.tuxmobil.fahrplan.congress.extensions.*
+
+import android.content.Context
+import android.view.ContextThemeWrapper
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.ImageView
+import android.widget.TextView
+
+import androidx.annotation.LayoutRes
+import info.metadude.android.eventfahrplan.commons.temporal.DateFormatter.Companion.newInstance
+
+import java.util.ArrayList
+import java.util.TreeSet
+
+import nerd.tuxmobil.fahrplan.congress.R
+import nerd.tuxmobil.fahrplan.congress.models.Session
+
+
+abstract class SessionsAdapter protected constructor(
+    context: Context?,
+    @LayoutRes layout: Int,
+    list: List<Session>?,
+    numDays: Int,
+    useDeviceTimeZone: Boolean
+) : ArrayAdapter<Session?>(
+    context!!, layout, list!!
+) {
+    companion object {
+        private const val TYPE_ITEM = 0
+        private const val TYPE_SEPARATOR = 1
+        private const val NUM_VIEW_TYPES = 2
+    }
+
+    @get:JvmName("getAdapterContext")
+    protected var context: Context? = null
+        private set
+
+    private val list: List<Session>?
+    private val numDays: Int
+    private var mMapper: ArrayList<Int>? = null
+    private var mSeparatorStrings: ArrayList<String>? = null
```

Comment:
Prefer interface types over concrete types, e. g. `MutableList<String>` -> `ArrayList<String>`. Please also check the other properties.

Reply:
How about the `TreeSet`?

Reply:
I overlooked that one, sorry. Should be fixed now